### PR TITLE
Alternative to atomic_add & Fix [!mayfail] Tests

### LINKED ISSUE #145: Alternative to atomic_add to track correspondences (dynfu::kinect_fusion_opencl_pose_estimation_pipeline_block)
Andrea confirmed what I determined experimentally: [`atomic_add`](https://www.khronos.org/registry/cl/sdk/1.1/docs/man/xhtml/atomic_add.html) is bad for GPU performance.

We still need some way to track/check correspondences to make sure we maintain tracking.  Perhaps a second reduce step where we simply count the number of correspondences?

Andrea doesn't think that tracking/checking the number of correspondences when running the pipeline live is particularly important, so perhaps we need some way to turn the tracking/checking on and off.

### LINKED ISSUE #154: dynfu::kinect_fusion_opencl_pose_estimation_pipeline_block [!mayfail] Tests
Address these tests: Make them pass.

### PR DESCRIPTION
**Pull request by RobertLeahy**
Closes #145
Closes #154

Fixes issues with `dynfu::kinect_fusion_opencl_measurement_pipeline_block` (see 700ee2eb633443014a96a5bb09ee4049b92db810).

### CODE REVIEW

File: src/kinect_fusion_opencl_pose_estimation_pipeline_block.cpp [hunk 1]
```diff
@@ -237,10 +255,29 @@ namespace dynfu {
...
+			Eigen::Matrix4f t_inc;
+			t_inc <<  1.0f, -gamma, beta, tx,
 					 gamma, 1.0f, -alpha, ty,
 				 	 -beta, alpha, 1.0f, tz,
 					 0.0f, 0.0f, 0.0f, 1.0f;
+			t_z.transposeInPlace();
+			t_z=t_inc*t_z;
+
+			//	Check count if applicable
+			if (cr) {
```
Comment (jordan-heemskerk) [inline src/kinect_fusion_opencl_pose_estimation_pipeline_block.cpp:267]: Do we want to check this before we tromp the `t_z`? We could put some logic in that doesn't update the transform for this frame and doesn't do the integration. The next consecutive frame should still be good enough to get the camera position again (ie. shouldn't invalidate the small angle assumptions). Just an idea. Don't have to put that logic in now, but I feel like we should be throwing that error before we solve/update the current transform.

Comment (RobertLeahy) [inline src/kinect_fusion_opencl_pose_estimation_pipeline_block.cpp:267]: Well we're throwing an exception, so we're bailing out: There is no continuing so destructive data modification isn't really relevant.

Given that this is exceptional we want to optimize for the non-throwing case, in which case we'll be computing the new `t_z` anyway.  Given that the GPU kernel which counts invalid correspondences can run asynchronously (and that the result can download asynchronously) it makes sense to do something else while we wait for it.

Comment (jordan-heemskerk) [inline src/kinect_fusion_opencl_pose_estimation_pipeline_block.cpp:267]: :+1: makes sense

**Comment by RobertLeahy** [comment]
@jordan-heemskerk: We good to merge this?

**Comment by jordan-heemskerk** [comment]
@RobertLeahy yea, fire away

**Pull request merged**